domain/src/main/java/no/difi/meldingsutveksling/receipt/Conversation.java

```diff
@@ -15,12 +16,61 @@
     @Id
     @GeneratedValue
     private String id;
-
     private String conversationId;
+    private String receiverIdentifier;
+    private String messageReference;
+    private String messageTitle;
+    private boolean pollable;
+    private ServiceIdentifier serviceIdentifier;
 
     @Cascade(CascadeType.PERSIST)
-    @OneToMany
-    List<MessageReceipt> messageReceipt;
+    @OneToMany(fetch = FetchType.EAGER)
+    List<MessageReceipt> messageReceipts;
+
+    Conversation() {}
+
+    private Conversation(String conversationId,
+                         String messageReference,
+                         String receiverIdentifier,
+                         String messageTitle,
+                         ServiceIdentifier serviceIdentifier,
+                         List<MessageReceipt> receipts) {
+        this.conversationId = conversationId;
+        this.messageReference = messageReference;
+        this.receiverIdentifier = receiverIdentifier;
+        this.messageTitle = messageTitle;
+        this.messageReceipts = receipts;
+        this.serviceIdentifier = serviceIdentifier;
+        switch (serviceIdentifier) {
+            case DPV:
+                this.pollable = true;
+            case DPI:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) End this switch case with an unconditional break, return or throw statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS128)